src/components/follow-link/follow-link.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { Link, SystemIcon, ESystemIconNames } from '@mdb/flora';
+import { useSession } from 'next-auth/react';
+import useSignInURL from '../../hooks/sign-in-url';
+import { Tag } from '../../interfaces/tag';
+import Tooltip from '../tooltip/tooltip';
+
+interface FollowLinkProps {
+    topic: Tag;
+    iconsOnly?: boolean;
+}
+
+const FollowLink: React.FunctionComponent<FollowLinkProps> = ({
+    topic,
+    iconsOnly = false,
+}) => {
+    const { status, data: session } = useSession();
+
+    const [showClickTooltip, setShowClickTooltip] = useState(false);
+
+    const [timeoutID, setTimeoutID] = useState<NodeJS.Timeout | null>(null);
+
+    const signInURL = useSignInURL();
+
+    const isLoggedIn = status === 'authenticated';
+    const followedTopics = session?.followedTopics as Tag[] | undefined;
+    const isFollowingAnyTopics = !!followedTopics && !!followedTopics.length;
+    const isFollowing = useMemo(
+        () =>
+            !!(
+                followedTopics &&
+                followedTopics.find(
+                    followedTopic => followedTopic.slug === topic.slug
+                )
+            ),
+        [followedTopics, topic.slug]
+    );
+
+    const hoverTooltipText = useMemo(() => {
+        if (!isLoggedIn) return 'Sign in to follow topics';
+
+        if (!isFollowingAnyTopics)
+            return 'Receive a monthly digest and recommended content based on topics you follow!';
+
+        if (iconsOnly)
+            return isFollowing ? 'Unfollow this topic' : 'Follow this topic';
+    }, [isLoggedIn, isFollowingAnyTopics, iconsOnly, isFollowing]);
+
+    // If we're not in a hoverTooltipText scenario, there should be no hover tooltip rendered.
+    const hasHoverTooltip = !!hoverTooltipText;
+
+    const onFollowClick = useCallback(() => {
+        let topics: Tag[];
+        // Never show the click tooltip when using the iconsOnly variant
+        if (!isFollowingAnyTopics) {
+            // TODO: Enter modal flow
+            alert('Onboarding modal will go here');
```

Comment:
if you're interested in at least incorporating the modal to show in scope of this, you can do the below now that https://github.com/mongodb/devcenter/pull/414 is added in main.

```
      openModal(
        <ScrollPersonalizationModal
          title={<>You&apos;re now following this topic. <br /> Interested in following other topics?</>}
           existingSelections={[...I think you'll want to add the previously followed topic in the prop here so it's already pre-selected in the tags]}
        />,
        { hideCloseBtn: true }
```

Reply:
but integration with the backend in the PUT's is not completed yet just FYI

Reply:
Planning to add that into this PR